src/handwritten/_install_daqmx.py

```diff
@@ -149,16 +90,16 @@ def _get_daqmx_installed_version() -> Optional[str]:
     elif sys.platform.startswith("linux"):
         try:
             _logger.debug("Checking for installed NI-DAQmx version")
+            commands_info = linux_commands[distro.id()]
+            query_command = commands_info["get_daqmx_version"]
+            query_output = subprocess.run(
+                query_command, stdout=subprocess.PIPE
+            ).stdout.decode("utf-8")
+
             if distro.id() == "ubuntu":
-                dpkg_output = subprocess.run(
-                    ["dpkg", "-l", "ni-daqmx"], stdout=subprocess.PIPE
-                ).stdout.decode("utf-8")
-                version_match = re.search(r"ii\s+ni-daqmx\s+(\d+\.\d+\.\d+)", dpkg_output)
+                version_match = re.search(r"ii\s+ni-daqmx\s+(\d+\.\d+\.\d+)", query_output)
             elif distro.id() == "opensuse" or distro.id() == "rhel":
-                rpm_output = subprocess.run(
-                    ["rpm", "-q", "ni-daqmx"], stdout=subprocess.PIPE
-                ).stdout.decode("utf-8")
-                version_match = re.search(r"ni-daqmx-(\d+\.\d+\.\d+)", rpm_output)
+                version_match = re.search(r"ni-daqmx-(\d+\.\d+\.\d+)", query_output)
```

Comment:
I think this is still too much distro-specific code here. I would wrap this in a method and put it in _linux_installation_commands.